cron_scripts/password_expiration_notification.pl

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use Getopt::Long;
+use JSON;
+use File::Slurp qw(read_file);
+use POSIX qw(strftime);
+use Pod::Usage qw(pod2usage);
+use Time::Piece;
+use FindBin;
+use Parallel::ForkManager;
+use LWP::UserAgent;
+
+=head1 NAME
+
+password_expiration_notification.pl - notify end users to change their password before expire
+
+=head1 SYNOPSIS
+
+  password_expiration_notification.pl [-help] [-debug] [-send_sms] [-pass_expiration_days <numExpDays>] [-notification_days <numNotiDays>] [-max_proc <maxNumProc>] -mail_file <mailFilePath> -sms_file <smsFilePath>
+
+=head1 DESCRIPTION
+
+This script allow us to detect in advance when a password is going to expire and notify end users that they need to change their password through Email and SMS.
+Mail and SMS content will have two placeholders {displayName} and {numDays} which will be replaced with actual value.
+{displayName} - user's display name
+{numDays} - number of days that password is going to expire
+
+=head1 OPTIONS
+
+=over
+
+=item -help
+
+Prints the help message.
+
+=item -pass_expiration_days I<numExpDays>
+
+Specifies number of days for password expiration limit (default is 90 days).
+
+=item -notification_days I<numNotiDays>
+
+Specifies notification period in days (default is 14 days).
+
+=item -mail_file I<mailFilePath>
+
+Specifies full file path, this file contains mail content to send for password notification.
+
+=item -sms_file I<smsFilePath>
+
+Specifies full file path, this file contains SMS content to send for password notification.
+
+=item -max_proc I<maxNumProc>
+
+Specifies maximum number of processes to be created to send mail/sms (default is 10).
+
+=item -debug
+
+Enable debug output.
+
+=item -send_sms
+
+Enable sending SMS.
+
+
+=back
+
+=cut
+
+use lib "$FindBin::Bin/lib";
+use ZCS::CustomAPI;
+
+my ( $Debug, $SendSMS );
+my $config_file = "$FindBin::Bin/config/config.json";
+
+my %opts   = process_options();
+my $config = process_config();
+
+my $notification_compare_days =
+  $opts{pass_expiration_days} - $opts{notification_days};
+my $pass_expiration_year_days = strftime( '%j',
+    localtime( time() - $opts{pass_expiration_days} * ( 24 * 60 * 60 ) ) );
+my $notification_timestamp =
+  strftime( '%Y%m%d000000',
+    localtime( time() - $notification_compare_days * ( 24 * 60 * 60 ) ) )
+  . "Z";
+
+my $soap_api = ZCS::CustomAPI->new(
+    conf => {
+        SOAPURI => "https://"
+          . $config->{MAILSTOREHOST}
+          . ":7071/service/admin/soap",
+        SOAPUser => $config->{MAILSTOREUSER},
+        SOAPPass => $config->{MAILSTOREPWD},
+    },
+    debug => $Debug
+);
+
+# search for accounts whose status are either of locked, lockout or pending
```

Comment:
This comment doesn't match the actual search/query being run.  Either remove the comment, or update to match the work being done.